client/assets/js/library.js

```diff
@@ -0,0 +1,122 @@
+document.addEventListener('DOMContentLoaded', function() {
+    initEventListeners();
+    targetElement.classList.add('show-img');
+    triangleLeft.addEventListener('click', function(){
+        handleArrowClickLeft();
+    });
+    triangleRight.addEventListener('click', function() {
+        handleArrowClickRight();
+    });
+});
+
+var touchStartCoords =  {'x':-1, 'y':-1}, // X and Y coordinates on mousedown or touchstart events.
+    touchEndCoords = {'x':-1, 'y':-1},// X and Y coordinates on mouseup or touchend events.
+    direction = 'undefined',// Swipe direction
+    minDistanceXAxis = 30,// Min distance on mousemove or touchmove on the X axis
+    maxDistanceYAxis = 30,// Max distance on mousemove or touchmove on the Y axis
+    maxAllowedTime = 1000,// Max allowed time between swipeStart and swipeEnd
+    startTime = 0,// Time on swipeStart
+    elapsedTime = 0,// Elapsed time between swipeStart and swipeEnd
+    elementArr = document.getElementsByClassName("story"), // Element to delegate
+    currentSpot = Math.floor(elementArr.length/2),
+    newSpot = null,
+    triangleLeft = document.querySelector('.triangle-left'),
+    triangleRight = document.querySelector('.triangle-right'),
+    targetElement = elementArr[currentSpot];
+    console.log('classList:', elementArr[currentSpot]);
+    console.log('current spot:', currentSpot);
+
+function moveLeft(){
+    currentSpot--;
```

Comment:
I think the glitchiness on the sides may be solved by changing this function slightly. You increment currentSpot-- at the beginning. But you may not want to do that. It makes it so that every time this function is called we increment to the left (in this case) which builds up excessive lefts past the edge of where we can be. Those all need to be recovered from. I think the following code might work better:

`function move(direction){
    //direction should be either:
    // 1 , for right
    //or
    // -1 , for left
    if(elementArr[currentSpot + direction] === undefined){
        triangleLeft.classList.remove('show-img');
        triangleLeft.classList.add('hide-img');
        return
    } else {
        triangleRight.classList.add('show-img');
    }
    currentSpot += direction;
    newSpot = elementArr[currentSpot];
    targetElement.classList.remove('show-img');
    targetElement.classList.add('hide-img');
    newSpot.classList.remove('hide-img');
    newSpot.classList.add('show-img');
    targetElement = newSpot;
    initEventListeners();
}`

You will likely notice that I changed the function name to just 'move' and added a parameter. We shouldn't need two functions with largely the same code. By passing in either 1 or -1 we can accomplish the same thing that the two functions were doing together.

Last note on this would be that you would need to update where the calls to this function were made so that you can pass in the right parameter and call the correct function name.

Reply:
Ugh, it didnt format the above function in a pretty way. Bummer.